### PR TITLE
K8SPS-93: Don't run multiple backups on the same pod

### PR DESCRIPTION
**Pull request by pooknull**
[K8SPS-93](https://jira.percona.com/browse/K8SPS-93) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPS-93

During backup, operator searches replica pods that are not running backup until it finds it. Otherwise, it sleeps for 10 seconds and searches again. If there are no free pods after the second search, multiply sleeping time by 2 (up to 10 minutes)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/e6dff6c/auto-config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/e6dff6c/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/e6dff6c/demand-backup.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/e6dff6c/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/e6dff6c/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/e6dff6c/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/e6dff6c/service-per-pod.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/e6dff6c/config.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/e6dff6c/scaling.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/e6dff6c/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/e6dff6c/limits.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/e6dff6c/users.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/170/commits/e6dff6ca2c71b5393bd127f372b535ab5e25d9c8
image: `perconalab/percona-server-mysql-operator:PR-170-e6dff6c`

File: build/run-backup.sh [hunk 1]
```diff
@@ -57,30 +57,49 @@ request_data() {
...
 request_backup() {
-	local http_code=$(
-		curl -s -o /dev/null \
-			-d "$(request_data)" \
-			-H "Content-Type: application/json" \
-			-w httpcode=%{http_code} \
-			"http://${SRC_NODE}:6033/backup/${BACKUP_NAME}" \
-			| sed -e 's/.*\httpcode=//'
-	)
+	readarray -td, SRC_NODES_ARR <<<"$SRC_NODES,"
```
Comment (hors) [inline build/run-backup.sh:60]: @pooknull more readable IFS=',' read -r -a SRC_NODES_ARR <<<"$SRC_NODES"

**Review by hors — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/67be2f3/auto-config.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/67be2f3/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/67be2f3/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/67be2f3/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/67be2f3/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/67be2f3/semi-sync.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/67be2f3/config.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/67be2f3/service-per-pod.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/67be2f3/scaling.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/67be2f3/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/67be2f3/limits.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/67be2f3/users.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/170/commits/67be2f3478fe8b6f91dc8e9d11af77f89682375d
image: `perconalab/percona-server-mysql-operator:PR-170-67be2f3`

File: build/run-backup.sh (hunk 1 above)
Comment (pooknull) [reply build/run-backup.sh]: https://github.com/percona/percona-server-mysql-operator/pull/170/commits/d49a543e823dd447f88e1846e1f70851cd3c1e69

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/d49a543/auto-config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/d49a543/init-deploy.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/d49a543/demand-backup.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/d49a543/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/d49a543/monitoring.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/d49a543/config.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/d49a543/scaling.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/d49a543/semi-sync.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/d49a543/users.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/d49a543/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/d49a543/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/d49a543/limits.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/170/commits/d49a543e823dd447f88e1846e1f70851cd3c1e69
image: `perconalab/percona-server-mysql-operator:PR-170-d49a543`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/auto-config.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/semi-sync.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/config.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/service-per-pod.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/scaling.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/limits.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/users.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/170/commits/4f7eed10fa6fe401d00ff7928e9decf9c795188b
image: `perconalab/percona-server-mysql-operator:PR-170-4f7eed1`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/auto-config.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/config.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/limits.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/4f7eed1/users.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/170/commits/4f7eed10fa6fe401d00ff7928e9decf9c795188b
image: `perconalab/percona-server-mysql-operator:PR-170-4f7eed1`

**Review by hors — CHANGES_REQUESTED** [review]
please check  demand-backup test

**Review by inelpandzic — COMMENTED** [review]
With this we can run multiple backup jobs in parallel on different nodes, is that good? Check the ticket to see @tplavcic comment on this: https://jira.percona.com/browse/K8SPS-93

File: cmd/sidecar/main.go [hunk 2]
```diff
@@ -53,6 +54,24 @@ type BackupConf struct {
...
+type Status struct {
+	isRunning atomic.Bool
+}
+
+func (s *Status) TryRunBackup() bool {
+	swapped := s.isRunning.CompareAndSwap(false, true)
+	if !swapped {
+		return false
+	}
+	return true
```
Comment (egegunes) [inline cmd/sidecar/main.go:68]:
```suggestion
	return s.isRunning.CompareAndSwap(false, true)
```

**Comment by egegunes** [comment]
@pooknull please fix shellcheck warnings

**Review by egegunes — CHANGES_REQUESTED** [review]

**Review by inelpandzic — COMMENTED** [review]
Maybe I'm not checking this correctly, but with 3 mysql instances I expect to see only 3 backups running in parallel. I run 5 backups and I saw this:
```
➜  percona-server-mysql-operator git:(dev/K8SPS-93) ✗ kp
NAME                                            READY   STATUS    RESTARTS   AGE
cluster1-mysql-0                                3/3     Running   0          13m
cluster1-mysql-1                                3/3     Running   1          11m
cluster1-mysql-2                                3/3     Running   1          9m49s
cluster1-orc-0                                  2/2     Running   0          13m
cluster1-orc-1                                  2/2     Running   0          12m
cluster1-orc-2                                  2/2     Running   0          11m
percona-server-mysql-operator-cf897d668-9zszv   1/1     Running   0          124m
xb-backup-1-s3-us-west-6krqs                    1/1     Running   0          49s
xb-backup-2-s3-us-west-9zd4m                    1/1     Running   0          38s
xb-backup-3-s3-us-west-v5wrz                    1/1     Running   0          28s
xb-backup-4-s3-us-west-b6gqx                    1/1     Running   0          20s
xb-backup-5-s3-us-west-j7wxj                    1/1     Running   0          13s
```
ps-backup:
```
➜  percona-server-mysql-operator git:(dev/K8SPS-93) ✗ k get ps-backup
NAME       STORAGE      DESTINATION                                                    STATE     COMPLETED   AGE
backup-1   s3-us-west   s3://operator-testing/inel/cluster1-2022-09-22-10:46:06-full   Running               57s
backup-2   s3-us-west   s3://operator-testing/inel/cluster1-2022-09-22-10:46:17-full   Running               46s
backup-3   s3-us-west   s3://operator-testing/inel/cluster1-2022-09-22-10:46:27-full   Running               36s
backup-4   s3-us-west   s3://operator-testing/inel/cluster1-2022-09-22-10:46:35-full   Running               28s
backup-5   s3-us-west   s3://operator-testing/inel/cluster1-2022-09-22-10:46:42-full   Running               21s
```

Is that correct?

**Comment by egegunes** [comment]
@inelpandzic what do you see in the logs? from what I understand by reading the code excess backups should sleep and retry until pod becomes available. am I right @pooknull ?

**Comment by inelpandzic** [comment]
> @inelpandzic what do you see in the logs? from what I understand by reading the code excess backups should sleep and retry until pod becomes available. am I right @pooknull ?

Yes, yes, I can see that in the code as well, but I expected not to see them like in the running state. But maybe my expectation was wrong.

**Review by inelpandzic — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| We run 0 out of 13|

commit: https://github.com/percona/percona-server-mysql-operator/pull/170/commits/02e37cca62a20e677977a2022e629966a2f72b84
image: `perconalab/percona-server-mysql-operator:PR-170-02e37cc`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/ca9ea8c/demand-backup.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/ca9ea8c/auto-config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/ca9ea8c/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/ca9ea8c/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/ca9ea8c/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/ca9ea8c/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/ca9ea8c/service-per-pod.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/ca9ea8c/config.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/ca9ea8c/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/ca9ea8c/limits.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/ca9ea8c/tls-cert-manager.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/ca9ea8c/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/ca9ea8c/users.log) |
| We run 13 out of 13|

commit: https://github.com/percona/percona-server-mysql-operator/pull/170/commits/ca9ea8c0b7bfa6b31a2c53aaa7d06c81585edc23
image: `perconalab/percona-server-mysql-operator:PR-170-ca9ea8c`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/9ce2831/demand-backup.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/9ce2831/auto-config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/9ce2831/init-deploy.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/9ce2831/config.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/9ce2831/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/9ce2831/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/9ce2831/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/9ce2831/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/9ce2831/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/9ce2831/limits.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/9ce2831/scaling.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/9ce2831/tls-cert-manager.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/9ce2831/users.log) |
| We run 13 out of 13|

commit: https://github.com/percona/percona-server-mysql-operator/pull/170/commits/9ce2831bdfaa2604c5aa9cc5eb8b44856fe75792
image: `perconalab/percona-server-mysql-operator:PR-170-9ce2831`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/03099cf/demand-backup.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/03099cf/auto-config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/03099cf/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/03099cf/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/03099cf/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/03099cf/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/03099cf/service-per-pod.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/03099cf/config.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/03099cf/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/03099cf/limits.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/03099cf/tls-cert-manager.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/03099cf/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/03099cf/users.log) |
| We run 13 out of 13|

commit: https://github.com/percona/percona-server-mysql-operator/pull/170/commits/03099cff2436d7ec91585207a4480729a2f184df
image: `perconalab/percona-server-mysql-operator:PR-170-03099cf`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/bef3ca3/init-deploy.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/bef3ca3/auto-config.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/bef3ca3/demand-backup.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/bef3ca3/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/bef3ca3/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/bef3ca3/semi-sync.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/bef3ca3/scaling.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/bef3ca3/service-per-pod.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/bef3ca3/users.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/bef3ca3/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/bef3ca3/limits.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/bef3ca3/tls-cert-manager.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/bef3ca3/config.log) |
| We run 13 out of 13|

commit: https://github.com/percona/percona-server-mysql-operator/pull/170/commits/bef3ca3146161be4afd6da711a43f9386cc277ad
image: `perconalab/percona-server-mysql-operator:PR-170-bef3ca3`

File: controllers/perconaservermysqlbackup_controller.go [hunk 3]
```diff
@@ -230,6 +230,10 @@ func (r *PerconaServerMySQLBackupReconciler) Reconcile(ctx context.Context, req
 			return rr, errors.Errorf("storage type %s is not supported", storage.Type)
 		}
 
+		status.Image = cluster.Spec.Backup.Image
+		status.SSLSecretName = cluster.Spec.SSLSecretName
```
Comment (egegunes) [inline controllers/perconaservermysqlbackup_controller.go:234]: what's the point of having this in status?

Comment (pooknull) [reply controllers/perconaservermysqlbackup_controller.go:234]: @egegunes It was added in https://github.com/percona/percona-server-mysql-operator/pull/158. We need this for `delete-backup` finalizer job when cluster doesn't exist

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/80696d6/demand-backup.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/80696d6/auto-config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/80696d6/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/80696d6/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/80696d6/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/80696d6/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/80696d6/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/80696d6/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/80696d6/limits.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/80696d6/scaling.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/80696d6/tls-cert-manager.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/80696d6/users.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-170/80696d6/config.log) |
| We run 13 out of 13|

commit: https://github.com/percona/percona-server-mysql-operator/pull/170/commits/80696d6e55bb8e31e1483fad23726403929868a6
image: `perconalab/percona-server-mysql-operator:PR-170-80696d6`

**Review by inelpandzic — APPROVED** [review]

**Review by egegunes — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Pull request merged**